### ISSUE TITLE
Sandbox and Production Modes for Projects

### ISSUE THREAD

**Issue by christad92** [OpenFn/lightning#2524]
Users should be able to switch between production and sandbox modes in their projects. 

Currently, users have to create two projects - (one for testing and the other for production) to achieve these two modes but it would be great to allow users to switch between these modes in their projects. At a project level, we will keep these 2 versions of credentials, project settings, and workflows. 

When adding project credentials, users should be prompted to add production and sandbox credentials. In some cases the sandbox credential and production credentials can be the same and users should be able to select a checkbox saying "Use this credential for production and sandbox mode".  

**Project settings**: Users should be able to configure GitHub connections and data storage for both modes.

**Workflow**: Every workflow will have production and sandbox modes. When changes are made to sandbox mode, users should be able to deploy them to production by clicking a deploy to production button. Before deploying changes to production, check that all sandbox variables are available for production. e.g. we should not have sandbox credentials and not have. Edits cannot be made in production mode, users should be asked to switch to sandbox mode before making edits. 

**History**: The history shown to users should be limited by the active mode of the user. A user in sandbox mode should only see runs in the sandbox mode. When in production mode, users should only see runs requested in the production mode. 

Only admins and owners can move a project from sandbox mode to production mode. 

Related to #2525 

See current process for handling sandbox projects: https://docs.google.com/document/d/1e-sYw2GUDqWfHoIgZ11ko06LQsKL5M9xzoNCaPgquUQ/edit?tab=t.0#heading=h.w8jezamffjxm

**Comment by taylordowns2000** [comment]
@theroinaochieng , @stuartc , @josephjclark , this is still very high priority, but blocked by #3007 and potential CLI deploy updates.

**Comment by taylordowns2000** [comment]
Noting that while the issue has changed quite a bit, this is in progress. It's only in the design phase, not the coding phase. We'll likely spin out a _lot_ of sub-issues from this one (cc @josephjclark )

**Comment by josephjclark** [comment]
Closing in favour of #3392
